atom/browser/ui/message_box_mac.mm

```diff
@@ -71,10 +71,14 @@ - (void)alertDidEnd:(NSAlert*)alert
 
   switch (type) {
     case MESSAGE_BOX_TYPE_INFORMATION:
-      [alert setAlertStyle:NSInformationalAlertStyle];
+      alert.alertStyle = NSAlertStyleInformational;
       break;
     case MESSAGE_BOX_TYPE_WARNING:
-      [alert setAlertStyle:NSWarningAlertStyle];
+    case MESSAGE_BOX_TYPE_ERROR:
+      // NSAlertStyleWarning shows the app icon while NSAlertStyleCritical
```

Comment:
Might be worth adding this to the docs just like the windows note over at https://github.com/electron/electron/blob/master/docs/api/dialog.md#dialogshowmessageboxbrowserwindow-options-callback

Just something like "On macOS 'error' and 'warning' result in the same dialog"